bin/main

```diff
@@ -1,99 +1,125 @@
 #!/usr/bin/env ruby
-def display_board(moves_array = [' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' '])
-  puts "#{moves_array[0]} | #{moves_array[1]} | #{moves_array[2]}"
-  puts '----------'
-  puts "#{moves_array[3]} | #{moves_array[4]} | #{moves_array[5]}"
-  puts '----------'
-  puts "#{moves_array[6]} | #{moves_array[7]} | #{moves_array[8]}"
-end
 
-puts 'Hi, Welcome to TicTacToe game'
+require_relative '../lib/board.rb'
+require_relative '../lib/player.rb'
 
-puts 'Insert name of players: '
+class TicTacToe
+  def create_players(name1, name2)
+    @players = [Player.new(name1), Player.new(name2)]
+  end
 
-players = []
-sym = []
+  def create_board
+    @board = Board.new
+  end
 
-players << gets.chomp.to_s
-players << gets.chomp.to_s
+  def print_board
+    print @board.display_board
+  end
 
-puts "Hi #{players[0]} and #{players[1]}, welcome to tictactoe game: "
+  def pick_symbol(sym1, sym2)
+    @players[0].pick_symbol(sym1)
+    @players[1].pick_symbol(sym2)
+  end
 
-print display_board
+  def set_beginner
+    beginner = gets.chomp.to_s
+    until [@players[0].name, @players[1].name].include?(beginner)
+      puts 'wrong input, Please insert correct player name: '
+      beginner = gets.chomp.to_s
+    end
 
-puts "Ok #{players[0]} please pick your symbol: "
+    @players[0], @players[1] = @players[1], @players[0] unless beginner == @players[0].name
+  end
 
-sym[0] = gets.chomp.to_s
+  def put_current_player(step)
+    @current_player = @players[step % 2]
+  end
 
-puts "Ok #{players[1]} please pick your symbol: "
+  def check_move(pick)
+    (@board.symbol[pick] != ' ') || !(0..10).cover?(pick) ? true : false
+  end
 
-sym[1] = gets.chomp.to_s
+  def update_board(pick)
+    @board.update_board(pick, @current_player)
+  end
+
+  def win_move?
+    @win_triplets = [[1, 2, 3], [4, 5, 6], [7, 8, 9], [1, 4, 7], [2, 5, 8], [3, 6, 9], [1, 5, 9], [3, 5, 7]]
+    @win_triplets.each do |triplet|
+      return true if [@board.symbol[triplet[0] - 1], @board.symbol[triplet[1] - 1],
+                      @board.symbol[triplet[2] - 1]].all?(@current_player.sym) == true
+    end
+  end
+end
```

Comment:
- [ ] All game-logic related resources (classes and/or modules) are kept in the `lib` directory. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone1/lib_dir_exist.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))